service/api_logic/games_logic.py

```diff
@@ -3,63 +3,28 @@
 from database.models import Games, Country, TeamIndex, League, Sport
 from dto.pagination import Pagination
 from exept.handle_exeptions import handle_exceptions
-from service.api_logic.scripts import apply_filters
 from sqlalchemy.orm import aliased
-from database.session import SessionLocal
 from logger.logger import Logger
+from service.api_logic.filter_manager.filter_manager_factory import FilterManagerFactory
 
-logger = Logger("logger", "all.log")
+class GamesService:
+    def __init__(self, games_dal):
+        self._games_dal = games_dal
+        self._logger = Logger("logger", "all.log").logger
 
-session = SessionLocal()
+    def get_games_today(self, filters_dto: GamesDTO(), pagination: Pagination):
 
-@handle_exceptions
-@logger.log_function_call()
-def get_games_today(
-        filters_dto: dict,
-        pagination: Pagination
-):
-    home_team = aliased(TeamIndex)
-    away_team = aliased(TeamIndex)
+        query = self._games_dal.get_query()
 
-    query = (
-        session.query(
-            Games,
-            League.name.label("league_name"),
-            League.logo.label("league_logo"),
-            Country.name.label("country_name"),
-            home_team.name.label("home_team_name"),
-            home_team.logo.label("home_team_logo"),
-            away_team.name.label("away_team_name"),
-            away_team.logo.label("away_team_logo"),
-            Games.score_home_team,
-            Games.score_away_team,
-            Games.status,
-            Games.time,
-            Games.date,
-            Games.api_id
-        )
-        .join(League, Games.league_id == League.league_id)
-        .join(Country, Games.country_id == Country.country_id)
-        .join(Sport, Games.sport_id == Sport.sport_id)
-        .join(home_team, Games.team_home_id == home_team.team_index_id)
-        .join(away_team, Games.team_away_id == away_team.team_index_id)
-    )
+        filtered_query = FilterManagerFactory.apply_filters(Games, query, filters_dto)
```

Comment:
Factory should be only providing object instances - not doing some logic. That violates Factory design pattern

and also, why do we need this `filtered_query` here? Why not let our FilterManager or whoever to just filter and return the objects we need already filtered and spare us the effort?

You shouldn't rewrite your `apply_filters` method, though, it could be reused, instead write another method which will return the actual objects of your filtered query.

Try achieving something like this:

filter_manager.apply_filters(Games, query, filters).execute()

where `execute()` will get your filtered objects

Not too sure about the name "execute" though, think of something